### PR TITLE
Fix reading empty Parquet DataPage

### PR DESCRIPTION
**Pull request by majetideepak**
Resolves the compression issue 7 here https://github.com/facebookincubator/velox/issues/9560

### CODE REVIEW

**Comment by majetideepak** [comment]
@yingsu00 can you take a look at this fix? Thanks!

**Comment by majetideepak** [comment]
@yingsu00 can you take a look at this PR?

**Comment by majetideepak** [comment]
@yingsu00, @nmahadevuni Can you review this fix as well?

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp [hunk 1]
```diff
@@ -1215,3 +1215,24 @@ TEST_F(ParquetReaderTest, testLzoDataPage) {
           .str(),
       "31232");
 }
+
+TEST_F(ParquetReaderTest, testEmptyDataPage) {
+  const std::string sample(getExampleFilePath("snappy.parquet"));
+
+  facebook::velox::dwio::common::ReaderOptions readerOptions{leafPool_.get()};
```
Comment (pedroerp) [inline velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp:1222]: nit: we should put the entire test suite under a facebook::velox::dwio::test` namespace so you don't need to specify the full path to everything.

Comment (majetideepak) [reply velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp]: I fixed this to `dwio::common::ReaderOptions`. There are a couple of `using namespace ...` on top. This is now consistent with the `velox/dwio/dwrf/test/ReaderTest.cpp` test.  Both are not under any namespace.

Comment (majetideepak) [reply velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp]: I cleaned up the remaining as well.

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp [hunk 2]
```diff
@@ -1215,3 +1208,24 @@ TEST_F(ParquetReaderTest, testLzoDataPage) {
           .str(),
       "31232");
 }
+
+TEST_F(ParquetReaderTest, testEmptyDataPage) {
+  const std::string sample(getExampleFilePath("snappy.parquet"));
```
Comment (yingsu00) [inline velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp:1213]: Maybe rename this file to empty_v2datapage.parquet?

**Review by yingsu00 — COMMENTED** [review]
@majetideepak Thanks for fixing the bug! It looks good, just one nit.

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp (hunk 2 above)
Comment (majetideepak) [reply velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp]: Fixed!

**Review by yingsu00 — APPROVED** [review]
Thanks @majetideepak !

**Review by Yuhta — APPROVED** [review]

**Pull request closed without merging**